In [61]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download(
    "vivovinco/20222023-football-player-stats",
    output_dir="../data/raw",
    force_download=True
    
    )

print("Path to dataset files:", path)

100%|██████████| 470k/470k [00:00<00:00, 2.02MB/s]

Extracting files...
Path to dataset files: ../data/raw


## 1. Notebook Objective

This notebook aims to understand the structure of the 2022–2023 football player statistics dataset, check data quality, and prepare the first observations for future analysis.


## 2. Library Imports

Import the main libraries used for data loading, manipulation, and basic analysis.


In [62]:
import pandas as pd

## 3. Dataset Loading

Load the dataset into a pandas DataFrame.

In [64]:
# Load the dataset
df = pd.read_csv("../data/raw/2022-2023 Football Player Stats.csv", encoding="latin1",sep=";")


## 4. First Rows Preview

Display the first rows to check if the dataset was loaded correctly.


In [65]:
# First lines of the dataset
df.head()

,Rk,Player,Nation,Pos,Squad,Comp,Age,Born,MP,Starts,...,Off,Crs,TklW,PKwon,PKcon,OG,Recov,AerWon,AerLost,AerWon%
0,1,Brenden Aaronson,USA,MFFW,Leeds United,Premier League,22,2000,20,19,...,0.17,2.54,0.51,0.0,0.0,0.00,4.86,0.34,1.19,22.2
1,2,Yunis Abdelhamid,MAR,DF,Reims,Ligue 1,35,1987,22,22,...,0.05,0.18,1.59,0.0,0.0,0.00,6.64,2.18,1.23,64.0
2,3,Himad Abdelli,FRA,MFFW,Angers,Ligue 1,23,1999,14,8,...,0.00,1.05,1.40,0.0,0.0,0.00,8.14,0.93,1.05,47.1
3,4,Salis Abdul Samed,GHA,MF,Lens,Ligue 1,22,2000,20,20,...,0.00,0.35,0.80,0.0,0.0,0.05,6.60,0.50,0.50,50.0
4,5,Laurent Abergel,FRA,MF,Lorient,Ligue 1,30,1993,15,15,...,0.00,0.23,2.02,0.0,0.0,0.00,6.51,0.31,0.39,44.4


## 5. Number of Rows and Columns

Check the dataset size using the number of rows and columns.


In [ ]:
# Dataset shape
df.shape

(2689, 124)

## 6. Data Types

Inspect the data types of each column to identify possible conversion needs.

In [58]:
# Dataset summary statistics
print("info about the dataset:\n")
df.info()

print("\ntypes of data in the dataset:")
df.dtypes


info about the dataset:

<class 'pandas.DataFrame'>
RangeIndex: 2689 entries, 0 to 2688
Columns: 124 entries, Rk to AerWon%
dtypes: float64(112), int64(7), str(5)
memory usage: 2.5 MB

types of data in the dataset:


Rk           int64
Player         str
Nation         str
Pos            str
Squad          str
            ...   
OG         float64
Recov      float64
AerWon     float64
AerLost    float64
AerWon%    float64
Length: 124, dtype: object

## 7. Available Columns

List all columns available in the dataset.


In [ ]:
#The available columns were listed to understand which variables can be used for player performance analysis, comparison, and future feature engineering.
columns = df.columns.tolist()

print(f"The dataset contains {len(columns)} columns:")
for column in columns:
    print("-", column)

The dataset contains 124 columns:
- Rk
- Player
- Nation
- Pos
- Squad
- Comp
- Age
- Born
- MP
- Starts
- Min
- 90s
- Goals
- Shots
- SoT
- SoT%
- G/Sh
- G/SoT
- ShoDist
- ShoFK
- ShoPK
- PKatt
- PasTotCmp
- PasTotAtt
- PasTotCmp%
- PasTotDist
- PasTotPrgDist
- PasShoCmp
- PasShoAtt
- PasShoCmp%
- PasMedCmp
- PasMedAtt
- PasMedCmp%
- PasLonCmp
- PasLonAtt
- PasLonCmp%
- Assists
- PasAss
- Pas3rd
- PPA
- CrsPA
- PasProg
- PasAtt
- PasLive
- PasDead
- PasFK
- TB
- Sw
- PasCrs
- TI
- CK
- CkIn
- CkOut
- CkStr
- PasCmp
- PasOff
- PasBlocks
- SCA
- ScaPassLive
- ScaPassDead
- ScaDrib
- ScaSh
- ScaFld
- ScaDef
- GCA
- GcaPassLive
- GcaPassDead
- GcaDrib
- GcaSh
- GcaFld
- GcaDef
- Tkl
- TklWon
- TklDef3rd
- TklMid3rd
- TklAtt3rd
- TklDri
- TklDriAtt
- TklDri%
- TklDriPast
- Blocks
- BlkSh
- BlkPass
- Int
- Tkl+Int
- Clr
- Err
- Touches
- TouDefPen
- TouDef3rd
- TouMid3rd
- TouAtt3rd
- TouAttPen
- TouLive
- ToAtt
- ToSuc
- ToSuc%
- ToTkl
- ToTkl%
- Carries
- CarTotDist
- CarPrgDist
- CarProg

## 8. Missing Values

Check if the dataset contains missing values.

In [ ]:
# Missing values were checked to identify incomplete columns and understand whether they may affect future analysis, cleaning, feature engineering, or modeling steps.
missing_values = df.isnull().sum()
missing_percentage = (missing_values / len(df)) * 100

missing_summary = pd.DataFrame({
    "missing_values": missing_values,
    "missing_percentage": missing_percentage
})

missing_summary[missing_summary["missing_values"] > 0].sort_values(
    by="missing_values",
    ascending=False
)

,missing_values,missing_percentage
Nation,1,0.037189


## 9. Duplicates

Check if the dataset contains duplicated rows.

In [54]:
duplicate_rows = df.duplicated().sum()

print(f"The dataset contains {duplicate_rows} duplicated rows.")

The dataset contains 0 duplicated rows.


## 10. First Observations About the Dataset

The dataset was successfully loaded and contains **2,689 rows** and **124 columns**.

The data is structured at the player level and includes identification columns, such as player name, nationality, position, squad, and competition, as well as several performance metrics related to minutes played, goals, shots, passing, assists, defensive actions, possession, and aerial duels.

Most columns were loaded as numerical values, which is useful for future statistical analysis, feature engineering, and modeling. A small number of columns were loaded as text, mainly categorical variables such as player, nation, position, squad, and competition.

Only one missing value was found, in the `Nation` column. No duplicated rows were found in the dataset.

Overall, the dataset appears to be well structured and suitable for future analysis. The next step will be to clean and standardize the data, validate column names and data types, and prepare the dataset for exploratory analysis and feature engineering.